#The MIT License
#
#Copyright (c) 2008 Heinz Wiesinger
#
#Permission is hereby granted, free of charge, to any person obtaining a copy
#of this software and associated documentation files (the "Software"), to deal
#in the Software without restriction, including without limitation the rights
#to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
#copies of the Software, and to permit persons to whom the Software is
#furnished to do so, subject to the following conditions:
#
#The above copyright notice and this permission notice shall be included in
#all copies or substantial portions of the Software.
#
#THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
#IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
#FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
#AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
#LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
#OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
#THE SOFTWARE.
--- vboxadd-timesync.sh	2008-06-25 20:09:33.000000000 +0200
+++ vboxadd-timesync.sh.new	2008-09-24 22:17:54.941071664 +0200
@@ -45,6 +45,9 @@
 elif [ -f /etc/gentoo-release ]; then
     system=gentoo
     PIDFILE="/var/run/vboxadd-timesync"
+elif [ -f /etc/slackware-version ]; then
+    system=slackware
+    PIDFILE="/var/run/vboxadd-timesync"
 else
     system=other
     if [ -d /var/run -a -w /var/run ]; then
@@ -123,6 +126,35 @@
     fi
 fi
 
+if [ "$system" = "slackware" ]; then
+    daemon() {
+	$1 $2
+    }
+
+    killproc() {
+        killall $1
+        rm -f $PIDFILE
+    }
+
+    fail_msg() {
+        echo " ...fail!"
+    }
+
+    succ_msg() {
+        echo " ...done."
+    }
+
+    status() {
+        echo -n "Checking for vboxadd-timesync"
+        if [ -f /var/run/$1 ]; then
+            echo " ...running"
+        else
+            echo " ...not running"
+        fi
+    }
+
+fi
+
 if [ "$system" = "other" ]; then
     fail_msg() {
         echo " ...fail!"
